python/lsst/ts/dimm/dimm_csc.py

```diff
@@ -237,10 +310,28 @@ async def seeing_loop(self):
 
         while self.seeing_loop_running:
             try:
-                data = await self.model.controller.get_measurement()
-                self.evt_dimmMeasurement.set_put(**data)
-            except Exception as e:
-                self.log.exception(e)
+                data = await self.controller.get_measurement()
+                if data is not None:
+                    self.evt_dimmMeasurement.set_put(**data)
+                await asyncio.sleep(self.heartbeat_interval)
+            except Exception:
+                self.log.exception("Error in seeing loop.")
+                self.fault(
+                    code=SEEING_LOOP_DONE,
+                    report="Error in seeing loop.",
+                    traceback=traceback.format_exc(),
+                )
+                break
+
+    def fault(self, code, report, traceback=""):
+        self.telemetry_loop_running = False
+        self.seeing_loop_running = False
+
+        try:
+            self.controller.stop()
```

Comment:
The problem now is that stop is a coroutine. So you need to handle it differently... :(